packages/sync-actions/src/utils/create-build-actions.js

```diff
@@ -1,13 +1,23 @@
-export default function createBuildActions (diff, doMapActions) {
+function applyOnBeforeDiff (before, now, fn) {
+  return fn && typeof fn === 'function' ? fn(before, now) : [before, now]
+}
+
+export default function createBuildActions (
+  differ, doMapActions, onBeforeDiff,
+) {
   return function buildActions (now, before, options = {}) {
     if (!now || !before)
       throw new Error('Missing either `newObj` or `oldObj` ' +
         'in order to build update actions')
 
-    // diff 'em
-    const diffed = diff(before, now)
+    const [
+      proccedBefore, processedNow,
+    ] = applyOnBeforeDiff(before, now, onBeforeDiff)
+
+    const diffed = differ(proccedBefore, processedNow)
```

Comment:
I renamed the arg. The library seems to call a `diff` the same as the `differ` creating a `diff` which confused me quite a bit with this argument shuffling.

Reply:
Just a nitpick: maybe you should rename the variable to `processedBefore`, and also on line [21](https://github.com/commercetools/nodejs/pull/304/files#diff-fb58f097aa83a7bd481dab8d7c44cf2aR21)